lib: libc: minimal: proper cast to "(char *)" from "(const char *)"

laroche opened the pull request:
The string "" is of type '(const char *)', so add a cast over to '(char *)' to clean up source code.

The change to stdout_console.c is purely cosmetic and just results in more compact written code.

keith-packard reviewed (changes requested):
The change to strerror.c is fine; I'd leave out the other changes as they aren't even vaguely related though.

laroche committed:
lib: libc: minimal: proper cast to "(char *)" from "(const char *)"

The string "" is of type '(const char *)', so add a cast over to
'(char *)' to clean up source code.

Signed-off-by: Florian La Roche <Florian.LaRoche@gmail.com>
